src/shared.ts

```diff
@@ -143,16 +143,23 @@ export const ESGlobalKeys = SetCreate([
     'Intl', // Unstable
 ]);
 
-// These are foundational things that should never be wrapped but are equivalent
+// These are intrinsics that can be reached by syntax, and must be mapped between realms.
 // TODO: revisit this list.
-export const ReflectiveIntrinsicObjectNames = [
-    'Object',
-    'Function',
-    'URIError',
-    'TypeError',
-    'SyntaxError',
-    'ReferenceError',
-    'RangeError',
-    'EvalError',
-    'Error',
-];
+export function extractUndeniableIntrinsics(globalObj: typeof globalThis): any[] {
+    return globalObj.eval(`[
+        ({}).constructor,
+        (_=>1).constructor,
+        [].constructor,
```

Comment:
adding array as well, in case of array subclassing, not sure what the implications are here.